tests/__tests__/sitka_module.test.ts

```diff
@@ -1,98 +1,134 @@
+    const actual = sitka.getModules().text.moduleName
+    expect(actual).toEqual("text")
   })
 
-  test('able to get modules', () => {
+  test('module is aware of other modules via this.modules property', () => {
+    const sitka = sitkaFactory()
+    const { text: textModule, color: colorModule } = sitka.getModules()
+    const expectedModulesValues = {
+      color: colorModule,
+      text: textModule,
+    }
+    expect(textModule.modules).toEqual(expect.objectContaining(expectedModulesValues))
   })
 
   test('reduxKey returns key', () => {
+    const sitka = sitkaFactory()
+    const { text: textModule } = sitka.getModules()
+    const actual = textModule.reduxKey()
+    expect(actual).toEqual("text")
   })
 
   test('setState (protected) updates redux state with handleText (public)', () => {
+    const { sitka, store } = createSitkaAndStore()
+    const { text: textModule } = sitka.getModules()
+    // Validates the state starts as default
+    const allState = store.getState()
+    const startingState = textModule.getStateTestDelegate(allState)
+    expect(startingState).toEqual(defaultTextModuleState)
+    // Validates the state updates using handleText, then calling protected setState
+    textModule.handleText(newTextModuleState)
+    const newAllState = store.getState()
+    const moduleState = textModule.getStateTestDelegate(newAllState)
+    expect(moduleState).toEqual(newTextModuleState)
   })
 
   test('resetState (protected) updates redux state to default with handleReset (public)', () => {
+    const { sitka, store } = createSitkaAndStore()
+    const { text: textModule } = sitka.getModules()
+    // Validates that we successfully change and start with an updated state
+    textModule.handleText(newTextModuleState)
+    const nonDefaultState = textModule.getStateTestDelegate(store.getState())
+    expect(nonDefaultState).toEqual(newTextModuleState)
+    // Validates we reset to default state
+    textModule.handleReset()
+    const currentState = textModule.getStateTestDelegate(store.getState())
+    textModule.getStateTestDelegate(store.getState())
+    expect(currentState).toEqual(defaultTextModuleState)
   })
 
   // SUBSCRIPTION
   test('subscriptions are created/provided with provideSubscriptions & createSubscription', () => {
+    const { sitka, store } = createSitkaAndStore()
+    const { text: textModule, color: colorModule } = sitka.getModules()
+    // Validates that we start with default state
+    const startingEditsState = textModule.getStateTestDelegate(store.getState()).numberOfEdits
+    expect(startingEditsState).toEqual(0)
+    // Validates that subscribed function is called and updates state
+    colorModule.handleColor("blue")
+    const updatedEditsState = textModule.getStateTestDelegate(store.getState()).numberOfEdits
+    expect(updatedEditsState).toEqual(1)
   })
 
   // FORKS
-  test('provideForks adds fork to Sitka', () => {
+  test('provideForks adds fork to Sitka', (done) => {
```

Comment:
This test, when switched to work with sitka factory shows an error referenceError: alert not defined